zcash_primitives/src/sapling/tree.rs

```diff
@@ -1,24 +1,28 @@
 use bitvec::{order::Lsb0, view::AsBits};
 use group::{ff::PrimeField, Curve};
-use incrementalmerkletree::Altitude;
+use incrementalmerkletree::{Hashable, Level};
 use lazy_static::lazy_static;
 use std::io::{self, Read, Write};
 
 use super::{
     note::ExtractedNoteCommitment,
     pedersen_hash::{pedersen_hash, Personalization},
 };
-use crate::merkle_tree::{HashSer, Hashable};
+use crate::merkle_tree::HashSer;
 
-pub const SAPLING_COMMITMENT_TREE_DEPTH: usize = 32;
-pub const SAPLING_COMMITMENT_TREE_DEPTH_U8: u8 = 32;
+pub const NOTE_COMMITMENT_TREE_DEPTH: u8 = 32;
+pub type CommitmentTree =
+    incrementalmerkletree::frontier::CommitmentTree<Node, NOTE_COMMITMENT_TREE_DEPTH>;
+pub type IncrementalWitness =
+    incrementalmerkletree::witness::IncrementalWitness<Node, NOTE_COMMITMENT_TREE_DEPTH>;
+pub type MerklePath = incrementalmerkletree::MerklePath<Node, NOTE_COMMITMENT_TREE_DEPTH>;
```

Comment:
Maybe these should have documentation.

Reply:
The `CommitmentTree` and `IncrementalWitness` type aliases are going to be removed entirely before the next release (because those types will no longer be used). `MerklePath` will stick around.

Reply:
I might actually be able to move these type aliases into `zcash_client_backend` now, but I don't think it's really worth it at the moment.